src/test/java/org/reaktivity/k3po/nukleus/ext/DuplexIT.java

```diff
@@ -318,6 +318,66 @@ public void shouldThrottleInitialOnlyServerSentData() throws Exception
 
     @Test
     @Specification({
+        "server.sent.throttle.stream/client",
+        "server.sent.throttle.stream/server"
+    })
+    public void shouldThrottleClientSentDataPerStream() throws Exception
+    {
+        k3po.finish();
+    }
+
+    // should pass because even if throttling is disabled, the receiver window is big enough to receive the message
+    @Test
+    @Specification({
+        "client.sent.throttle.off/client",
+        "client.sent.throttle.off/server"
+    })
+    public void shouldReceiveMessageThrottleOff() throws Exception
+    {
+        k3po.finish();
+    }
+
+    // should fail because the receiver window is not big enough and window adjustment is disabled
+    @Test
+    @Specification({
+        "client.sent.throttle.off.negative/client",
+        "client.sent.throttle.off.negative/server"
+    })
+    public void shouldNotReceiveMessageThrottleOff() throws Exception
+    {
+        thrown.expect(anyOf(isA(ComparisonFailure.class),
+                hasProperty("failures", hasItem(isA(ComparisonFailure.class)))));
+        k3po.finish();
+    }
+
+    // should fail (please see readme)
+    @Test
+    @Specification({
+        "server.set.window0.throttle.on/client",
+        "server.set.window0.throttle.on/server"
+    })
+    public void shouldReceiveMessageWindow0ThrottleingOn() throws Exception
+    {
+        thrown.expect(anyOf(isA(ComparisonFailure.class),
+                hasProperty("failures", hasItem(isA(ComparisonFailure.class)))));
+        k3po.finish();
+    }
+
+    // should fail because the receiver window is 0 and window adjustment is disabled
+    @Test
+    @Specification({
+        "server.set.window0.throttle.off/client",
+        "server.set.window0.throttle.off/server"
+    })
+    public void shouldNotReceiveMessageWindow0ThrottleingOff() throws Exception
```

Comment:
Like shouldNotReceiveMessageThrottleOff this test is a duplicate of server.sent.overflow and should be removed. There's nothing special about setting window to 0, it's just another value, There no more reason to test it than to test say value 756. If we had defined special behavior for window 0 (like not sending a window frame for that case) then it would be worth testing, but we have not, it just causes us to send a window frame with 0 bytes.

Reply:
Allright, removed it.